pkg/providers/subnet/subnet.go

```diff
@@ -252,35 +254,36 @@ func (p *DefaultProvider) minPods(instanceTypes []*cloudprovider.InstanceType, r
 	pods, _ := lo.MinBy(filteredInstanceTypes, func(i *cloudprovider.InstanceType, j *cloudprovider.InstanceType) bool {
 		return i.Capacity.Pods().Cmp(*j.Capacity.Pods()) < 0
 	}).Capacity.Pods().AsInt64()
-	return pods
+	//nolint:gosec
+	return int32(pods)
```

Comment:
We're doing a lot of this -- do we have a plan to come back to these and see how we can drop these `//nolint`s long-term?

Reply:
So we did discuss this. The nolints where added in any place where the v2 change to int32 caused overflow errors for the linter and I could not fix it. We made sure they were not anywhere where overflow was actually a concern. I am not sure if there is a fix we can implement for this but I am open to looking at it again.